File: src/models/common/RangeBand.ts
import {Option} from "../../components/common/InputSelectField";

export enum RangeBand {
    Engaged = 'Engaged',
    Short = 'Short',
    Medium = 'Medium',
    Long = 'Long',
    Extreme = 'Extreme',
    Strategic = 'Strategic'
}

export const getRangeOptions = (): Option[] => {
    return Object.values(RangeBand).map((value) => ({value}))
}

File: server/src/main/java/com/github/genraven/genesys/domain/RangeBand.java
package com.github.genraven.genesys.domain;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum RangeBand {
    ENGAGED("Engaged"),
    SHORT("Short"),
    MEDIUM("Medium"),
    LONG("Long"),
    EXTREME("Extreme"),
    STRATEGIC("Strategic");

    @JsonValue
    private final String label;
}
